### PR TITLE
Handle Python differences on Fedora

### PR DESCRIPTION
**Pull request by tachoknight**
Fedora's installation of Python gets the result of `distutils.sysconfig.get_python_lib(True,False)` as `lib64`, which causes a failure later on in the installation process as the install script is looking for the `python2.7/site-packages` directory in `lib`. This is not an issue with Ubuntu so the change tests for whether we're running on Linux, and if so and we're on Fedora, run `distutils.sysconfig.get_python_lib(False, False)` to return the same path as the rest of the Python files being built.

### CODE REVIEW

File: scripts/Python/finishSwigPythonLLDB.py [hunk 1]
```diff
@@ -649,7 +650,13 @@ def get_framework_python_dir_windows(vDictArgs):
             vDictArgs["--cmakeBuildConfiguration"])
 
     if strPythonInstallDir.__len__() != 0:
-        strWkDir = get_python_lib(True, False, strPythonInstallDir)
+        if platform.system() == 'Linux':
```
Comment (dcci) [inline scripts/Python/finishSwigPythonLLDB.py]: Please move this to a separate function.

**Review by dcci — CHANGES_REQUESTED** [review]

**Comment by dcci** [comment]
Please rebase and squash this in a single commit.
I think this hack is alright, but I'll run the tests to make sure this doesn't break other Linux(es).
I'd wait if anybody else has opinions on this, otherwise merge
cc: @fredriss / @adrian-prantl / @jasonmolenda / @jimingham / @JDevlieghere / @vedantk

**Comment by dcci** [comment]
@swift-ci please test linux

**Comment by tachoknight** [comment]
Per dcci's request, I moved the code into a function and put it near the other platform-specific functions.

**Review by dcci — APPROVED** [review]

**Comment by dcci** [comment]
Please rebase this as it contains a bunch of other unrelated changes now.
Otherwise, LGTM, and I'll merge it.

**Comment by dcci** [comment]
This looks still really weird (there are 40 commits in this pull request).

**Comment by tachoknight** [comment]
Yes, sorry about that, trying to fix it.

**Comment by tachoknight** [comment]
Fixed with the help of the friendly folks on the #git irc channel (thank you preaction and tkazi)

**Comment by dcci** [comment]
@swift-ci please test

**Comment by dcci** [comment]
@tachoknight I'll merge this if it passes CI. Thanks for your contribution.

**Comment by dcci** [comment]
@swift-ci please test

**Pull request merged**